diff -ur keepalived-1.1.20/keepalived/vrrp/vrrp.c keepalived-1.1.20.fixed/keepalived/vrrp/vrrp.c
--- keepalived-1.1.20/keepalived/vrrp/vrrp.c	2012-03-16 22:59:44.170925903 +0000
+++ keepalived-1.1.20.fixed/keepalived/vrrp/vrrp.c	2012-03-23 04:27:41.840913404 +0000
@@ -830,7 +830,7 @@
 		return 0;
 	} else if (hd->priority > vrrp->effective_priority ||
 		   (hd->priority == vrrp->effective_priority &&
-		    ntohl(iph->saddr) > VRRP_PKT_SADDR(vrrp))) {
+		    ntohl(iph->saddr) > ntohl(VRRP_PKT_SADDR(vrrp)))) {
 		log_message(LOG_INFO,
 		       "VRRP_Instance(%s) Received higher prio advert",
 		       vrrp->iname);
